src/System.Text.Json/docs/ReferenceHandling_spec.md

```diff
@@ -0,0 +1,433 @@
+
+When using `ReferenceLoopHandling.Ignore`, other objects that were already seen on the current graph branch will be ignored on Serialization.
+
+When using `PreserveReferencesHandling.All` you are signaling that your resulting JSON will contain *metadata* properties `$ref`, `$id` and `$values` which are going to act as reference identifiers (`$id`) and pointers (`$ref`). 
+Now, to read back those references, you have to use `MetadataPropertyHandling.Default` to indicate that *metadata* is expected in the payload passed to the Deserialize method.
+
+
+# Proposal
+
+```cs
+namespace System.Text.Json.Serialization
+{
+    /// <summary>
+    /// This class defines the various ways the <see cref="JsonSerializer"/> can deal with references on Serialization and Deserialization.
+    /// </summary>
+    public sealed class ReferenceHandling
+    {
+        //built-in options.
+        public static ReferenceHandling Default => new ReferenceHandling(PreserveReferencesHandling.None);
+        public static ReferenceHandling Preserve => new ReferenceHandling(PreserveReferencesHandling.All);
+
+        // For future, change to public if needed.
+        internal PreserveReferencesHandling PreserveHandlingOnSerialize => _preserveHandlingOnSerialize;
+        internal PreserveReferencesHandling PreserveHandlingOnDeserialize => _preserveHandlingOnDeserialize;
+        internal ReferenceLoopHandling LoopHandling => _loopHandling;
+
+        private PreserveReferencesHandling _preserveHandlingOnSerialize;
+        private PreserveReferencesHandling _preserveHandlingOnDeserialize;
+        private ReferenceLoopHandling _loopHandling;
+
+        public ReferenceHandling(PreserveReferencesHandling handling) : this(handling, handling, ReferenceLoopHandling.Error) { }
+
+        // For future, someone may want to define their own custom Handler with different behaviors of PreserveReferenceHandling on Serialize vs Deserialize and another ReferenceLoopHandling, such as ignore if added in a future.
+        private ReferenceHandling(PreserveReferencesHandling preserveHandlingOnSerialize, PreserveReferencesHandling preserveHandlingOnDeserialize, ReferenceLoopHandling loopHandling)
+        {
+            _preserveHandlingOnSerialize = preserveHandlingOnSerialize;
+            _preserveHandlingOnDeserialize = preserveHandlingOnDeserialize;
+            _loopHandling = loopHandling;
+        }
+    }
+
+    internal enum ReferenceLoopHandling
+    {
+        Error = 0,
+        // For future if requested by the community.
+        // Ignore = 1,
+    }
+
+    public enum PreserveReferencesHandling
+    {
+        None = 0,
+        All = 1,
+        // For future if requested by the community.
+        // Objects = 2,
+        // Arrays = 3,
+    }
+}
+
+namespace System.Text.Json
+{
+    public partial class JsonSerializerOptions
+    {
+        public ReferenceHandling ReferenceHandling { get; set; } = ReferenceHandling.Default;
```

Comment:
Explicitly call out the semantics of the default behavior for both serialize and deserialize (and how it isn't changing).

Reply:
What happens by default when we deserialize JSON containing metadata properties? Also, is there a way to ignore them?

Reply:
> What happens by default when we deserialize JSON containing metadata properties?

They are treated as any other property, if they are not mapped to a C# property (by using attribute `[JsonPropertyName("$id")]`, for example), they will be added to the `JsonExtensionData` dictionary if there is any.

> Is there a way to ignore them?

If by "ignore them" you mean completely skip them when parsing the JSON; no, there is no way, nor reason to do that. Newtonsoft.Json is also incapable of doing that.